cmd/run/run_test.go

```diff
@@ -870,6 +870,7 @@ func testServerMetricsReporting(t *testing.T, engine string) {
 	require.Contains(t, stringBody, "list_objects_further_eval_required_count")
 	require.Contains(t, stringBody, "list_objects_no_further_eval_required_count")
 	require.Contains(t, stringBody, "go_sql_idle_connections")
```

Comment:
Turns out these assertions are a bit misleading. `promauto` automatically registers and emits the zero value for a metric. Even if a metric doesn't have an observed value prometheus will report it. Just by virtue of having

```
var myCounter = promauto.NewCounter(...)
```
the metric will show up. It doesn't mean you ever call `myCounter.Inc()`, for example.

So this test is a bit of a misleading test. It's not actually making sure the metric usage is correct, it's just making sure the metric is somehow defined. Not sure if we care 🤷 ? The problem could be that we break code that stops reporting the value but this test still passes because the metric is still defined but not used correctly.

Reply:
@miparnisari @jpadilla do you two have an opinion here?

Reply:
This is fine as is